src/transport/manager/mod.rs

```diff
@@ -939,22 +942,50 @@ impl TransportManager {
 
                         return Ok(ConnectionEstablishedResult::Reject);
                     }
-                    None => {
-                        tracing::debug!(
+                    None => match dial_record.take() {
+                        Some(record)
+                            if record.connection_id() == &Some(endpoint.connection_id()) =>
+                        {
+                            tracing::debug!(
+                                target: LOG_TARGET,
+                                ?peer,
+                                connection_id = ?endpoint.connection_id(),
+                                address = ?endpoint.address(),
+                                "dialed connection opened as secondary connection",
+                            );
+
+                            context.secondary_connection = Some(AddressRecord::new(
+                                &peer,
+                                endpoint.address().clone(),
+                                SCORE_DIAL_SUCCESS,
+                                Some(endpoint.connection_id()),
+                            ));
+                        }
+                        None => {
+                            tracing::debug!(
+                                target: LOG_TARGET,
+                                ?peer,
+                                connection_id = ?endpoint.connection_id(),
+                                address = ?endpoint.address(),
+                                "secondary connection",
+                            );
+
+                            context.secondary_connection = Some(AddressRecord::new(
+                                &peer,
+                                endpoint.address().clone(),
+                                SCORE_DIAL_SUCCESS,
```

Comment:
Is it actually an incoming connection in this case, as `dial_record` is `None`?
`SCORE_*DIAL*_SUCCESS` confuses me a little here.

Reply:
It is. AFAIR there were at least one other `SCORE_SOMETHING_SUCCESS` but it got removed for reasons I can't remember. Its name should probably be changed to something else

Reply:
I renamed both of them to `SCORE_CONNECT_*` for consistence, I can revert the other rename if you don't like it.

Reply:
It's fine, thank you!